flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/iterator/RocksSingleStateIterator.java

```diff
@@ -23,11 +23,13 @@
 
 import javax.annotation.Nonnull;
 
+import java.io.Closeable;
+
 /**
  * Wraps a RocksDB iterator to cache it's current key and assigns an id for the key/value state to
  * the iterator. Used by {@link RocksStatesPerKeyGroupMergeIterator}.
  */
-class RocksSingleStateIterator implements AutoCloseable {
+class RocksSingleStateIterator implements Closeable {
```

Comment:
is this change necessary? The `AutoCloseable` is the newer interface.

Reply:
I made it `Closeable` because the `RocksStatesPerKeyGroupMergeIterator` now uses a `CloseableRegistry` which requires things to be `Closeable`. 

Alternatively, I could change `CloseableRegistry` to take `AutoCloseables` but I felt that would be the bigger change.

Reply:
got it, I think that's fine